File: 205. Isomorphic Strings/Python/Solution.py
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False
        pattern = {}
        exist = {}
        for i in range(0, len(s)):
            s_in = s[i] in exist
            t_in = t[i] in pattern
            if s_in and t_in:
                if ord(t[i]) + pattern[t[i]] != ord(s[i]):
                    return False
            elif not s_in and not t_in:
                pattern[t[i]] = ord(s[i]) - ord(t[i])
                exist[s[i]] = True
            else:
                return False
        return True

File: 205. Isomorphic Strings/Java/Solution.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        int[] m1 = new int[256];
        int[] m2 = new int[256];
        for (int i = 0; i < s.length(); i++) {
            char k1 = s.charAt(i);
            char k2 = t.charAt(i);
            if (m1[k1] != m2[k2]) {
                return false;
            }
            m1[k1] = i + 1;
            m2[k2] = i + 1;
        }
        return true;
    }
}